assets/src/components/detoursTable.tsx

```diff
@@ -65,83 +78,117 @@ export const DetoursTable = ({
+                      value={filter}
+                      onBlur={() =>
+                        fullStoryEvent("Detour Intersection Filter Used", {})
+                      }
+                      onChange={(e) => setFilter(e.target.value)}
+                      className="c-detour-list-filter__input"
+                    />
+                    {filter.length > 0 && (
+                      <button
+                        className="c-detour-list-filter__clear"
+                        onClick={() => setFilter("")}
+                        title="Clear"
+                      >
+                        <CircleXIcon />
+                      </button>
+                    )}
                   </div>
+                  <button
+                    type="submit"
+                    title="Submit"
+                    className="c-detour-list-filter__submit"
+                    onClick={setDebouncedFilter.bind(null, filter)}
+                    disabled={filter.length === 0}
+                  >
+                    <SearchIcon />
+                    Search
+                  </button>
                 </div>
-              </th>
-              <th className="px-3 py-3"></th>
-            </tr>
-          )}
-          <tr>
-            <th className="px-3 py-4">Route and direction</th>
-            <th className="px-3 py-4 u-hide-for-mobile">
-              Starting Intersection
+              </div>
             </th>
-            <th className="px-3 py-4 u-hide-for-mobile">
-              {timestampLabelFromStatus(status)}
+            <th className="px-3 py-3 text-end">
+              <Button
+                className="icon-link"
+                variant="outline-primary"
+                data-fs-element="Reset detour search"
+                type="button"
+                title="Clear Search"
+                onClick={resetInputs}
+              >
+                <XSquare />
+                Clear
+              </Button>
             </th>
-            {status === DetourStatus.Active && (
-              <th className="px-3 py-4 u-hide-for-mobile">Est. Duration</th>
-            )}
           </tr>
-        </thead>
-        <tbody>
-          {filteredData.length ? (
-            <PopulatedDetourRows
-              status={status}
-              data={filteredData}
-              onOpenDetour={onOpenDetour}
-            />
-          ) : (
-            <EmptyDetourRows message={`No ${status} detours.`} />
+        )}
+        <tr>
+          <th className="px-3 py-4">Route and direction</th>
+          <th className="px-3 py-4 u-hide-for-mobile">Starting Intersection</th>
+          <th className="px-3 py-4 u-hide-for-mobile">
+            {timestampLabelFromStatus(status)}
+          </th>
+          {status === DetourStatus.Active && (
+            <th className="px-3 py-4 u-hide-for-mobile">Est. Duration</th>
           )}
-        </tbody>
-      </>
+        </tr>
+      </thead>
+      <tbody>
+        {data.length ? (
```

Comment:
was this intentional to show no rows and not the no Past detours message when the intersection filter isn't returning any results?